### ISSUE TITLE
[REQUEST] Refining 3rd Party Cookies Management

### ISSUE THREAD

**Issue by Kraxys** [sereneblue/chameleon#180]
Currently Chameleon's 3rd Party Cookies Management exactly replicates what Firefox offer in that area. You could either allow all 3rd party cookies, or allow 3rd party from visited domains, or reject only 3rd party cookies related to known trackers and accept all others, or block them all.

But sometime you will be (in order to access to some site's functionality) to allow all 3rd party cookie, or at least to block only those of them not detected as being sent by a known tracker, and you could be uncomfortable with that  cookie mandatory sent on your browser. 

But there is an other (and complementary) way to handle that:  When you can't prevent a cookie from entering the browser, you can nevertheless prevent this cookie from leaving the browser.

Such a feature seems to be implemented in uMatrix, but it is overkill to have to use uM just for this feature, especially when already using uBlokO in "advanced mode", case in which it is not recommended to use uM alongside .

So maybe it could make sense to add such a feature in Chameleon in the future. It would add no benefit when 3rd party Cookies preferences is set to "Block All", but in others case, it could add a level of privacy.

Currently using Chameleon 0.12

**Comment by sereneblue** [comment]
If I'm understanding your request correctly, you're asking for granular control of which cookies are blocked similar to how uMatrix handles it? If so, that's quite a large of feature to add and falls outside of the scope of this addon. I only included the cookies handling option to expose the options for users because it was available.

**Comment by Kraxys** [comment]
> 
> 
> If I'm understanding your request correctly, you're asking for granular control of which cookies are blocked similar to how uMatrix handles it? If so, that's quite a large of feature to add and falls outside of the scope of this addon. I only included the cookies handling option to expose the options for users because it was available.

No, not granular in the "per domain" sense, I understand that would fall largely outside the Chameleon's scope. What I'm thinking about is an additional way to **globally** handle cookies.  uMatrix does that but a gazillion other things, and I'm speaking about the precise following feature, explained by Gorhill:
"_Blacklisted cookies are not prevented by uMatrix from entering your browser. However they are prevented from leaving your browser, which is what really matters_."
https://github.com/gorhill/uMatrix/wiki/Cookies

But maybe I'm misinterpreting what Gorhill said above??

**Comment by sereneblue** [comment]
I don't think that solution would be viable. For example, if you login to a website, cookies are set for that session. You wouldn't be able to maintain a session because the cookies aren't being sent back to the server. 

I'm not sure if uMatrix's cookie blocking actually works? I tried testing with http://www.whatarecookies.com/cookietest.asp but it still shows the cookie. I used the latest dev version (1.3.17rc0).

I think your best option is to use Temporary Containers or an extension made specifically to handle cookies.

**Comment by Kraxys** [comment]
> 
> 
> I don't think that solution would be viable. For example, if you login to a website, cookies are set for that session. You wouldn't be able to maintain a session because the cookies aren't being sent back to the server.
> 
Ok, what I wanted to say whas indeed incomplete. I meant "globally preventing **third party cookies** from leaving the browser". 

> I'm not sure if uMatrix's cookie blocking actually works? I tried testing with http://www.whatarecookies.com/cookietest.asp but it still shows the cookie. I used the latest dev version (1.3.17rc0).

Don't forget: The cookies are not supposed to be prevented from _entering_, but from _leaving_.

But actually I didn't perform any test concerning this uMatrix feature. My idea was just founded on what I saw what Gorhill wrote. 

> I think your best option is to use Temporary Containers or an extension made specifically to handle cookies.

It's surely an option fore sure. Just that I don't know any cookies managing addon offering to "prevent third party cookies from leaving the browser" (except of course uMatrix, I mean, but using this one just for that is a bit overkill).

But my purpose is absolutely not to t bother you with that. I just suggested it because I found this way of handling 3rd party cookies was smart and could offer a completion of the others ways of handling them (block all/Block non visited only/block non tracking only). If you think this feature is not so smart, or really outside the Chameleon scopes, no problem. Chameleon is very good already. :)
